### ISSUE TITLE
Poor first push user experience

### ISSUE THREAD

**Issue by sergiimk** [kamu-data/kamu-cli#450]
### Describe the bug

Pushing a new dataset into ODF repository is hard!

### Steps To Reproduce

I just created a new dataset in my local workspace. To push it to ODF repo I currently need:
- `kamu login -s https://platform.demo.kamu.dev`
- `kamu push my.dataset --to "odf+https://api.demo.kamu.dev/sergiimk/my.dataset"`

### Current Behavior

Note that:
1. I need to figure out the backend hostname myself - I cannot push to the same domain I logged in to
2. I need to specify `odf+http` protocol which most people don't know about
3. I need to provide my username and target dataset name in URL

### Expected Behavior

Ideally the following interactions "just work":
```
kamu login platform.demo.kamu.dev  # defaults to https scheme

# If only one known repository
kamu push my.dataset

# To disambiguate between multiple repositories
kamu push my.dataset --to platform.demo.kamu.dev
```

In other words:
- `kamu login` command should automatically register a repo alias: `kamu repo add <frontend domain> <backend url>`
- `kamu push my.dataset` should automatically pick a repo is there is only one
- `kamu push my.dataset --to platform.demo.kamu.dev` should understand that remote is using ODF protocol and construct a proper push URL for me

### kamu system info

```sh
-
```

**Comment by sergiimk** [comment]
This was auto-closed by the merge of CLI changes. Will re-open this until:
- Node is released with new endpoints
- New Jupyter image is release
- All components are deployed to demo env and tested

**Comment by sergiimk** [comment]
Following issues identified during the acceptance testing:
- https://github.com/kamu-data/kamu-cli/issues/946
- https://github.com/kamu-data/kamu-cli/issues/947

**Comment by sergiimk** [comment]
Conducted acceptance testing and all works as expected!
Closing as done :partying_face: :champagne:
